### PR TITLE
Add required_approving_review_count #89 (required go-github update)

### LINKED ISSUE #89: [Feature Request] support `required_approving_review_count`
### Affected Resource(s)
Please list the resources as a list, for example:
- github_branch_protection

### Important Factoids
Some teams are large enough that they want multiple reviewers approving it before allowing it to merge.

It is available via the [rest api](https://developer.github.com/v3/repos/branches/#update-branch-protection) but I do not see it yet in the upstream go library](https://github.com/google/go-github/blob/v15.0.0/github/repos.go) so we are blocked until that is added. I have created https://github.com/google/go-github/issues/905 to track.

image

### PR DESCRIPTION
**Pull request by quantumew**
* Update go-github to latest (0.18.2) for google/go-github#880
* Add new required_approving_review_count property
* Correct concomitant breaking changes

Fixes #89

### CODE REVIEW

**Comment by quantumew** [comment]
@radeksimko Hey! no rush or anything but I opened this PR. Don't let the XXL scare you from reviewing it, it is very small except it contains an update to the Github SDK. ACC tests passed locally and we implemented this change in our stack and it works great! Thanks for any input!!

File: github/resource_github_branch_protection.go [hunk 1]
```diff
@@ -96,6 +96,21 @@ func resourceGithubBranchProtection() *schema.Resource {
 							Type:     schema.TypeBool,
 							Optional: true,
 						},
+						"required_approving_review_count": {
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Description: "Number of approvals required to merge a pull request",
+							Default:     1,
+							// Old version of terraform, could not utilize validation.IntBetween
+							ValidateFunc: func(i interface{}, k string) (s []string, es []error) {
```
Comment (robsonpeixoto) [inline github/resource_github_branch_protection.go]: You can use:
```go
import "github.com/hashicorp/terraform/helper/validation"
...

ValidateFunc: validation.IntBetween(1, 6),
```

File: github/resource_github_branch_protection.go [hunk 2]
```diff
@@ -205,7 +220,7 @@ func resourceGithubBranchProtectionRead(d *schema.ResourceData, meta interface{}
 	}
 
 	if err := flattenAndSetRequiredPullRequestReviews(d, githubProtection); err != nil {
-		return fmt.Errorf("Error setting required_pull_request_reviews: %v", err)
+		return fmt.Errorf("Error setting required_pull_reuest_reviews: %v", err)
```
Comment (robsonpeixoto) [inline github/resource_github_branch_protection.go]: typo: `required_pull_request_reviews`

File: github/resource_github_branch_protection.go (hunk 1 above)
Comment (robsonpeixoto) [reply github/resource_github_branch_protection.go]: Will be needed to add this on `vendor.json`
```json
		{
			"checksumSHA1": "nEC56vB6M60BJtGPe+N9rziHqLg=",
			"path": "github.com/hashicorp/terraform/helper/validation",
			"revision": "41e50bd32a8825a84535e353c3674af8ce799161",
			"version": "v0.11.7",
			"versionExact": "v0.11.7"
		},
```

Comment (quantumew) [reply github/resource_github_branch_protection.go]: Ah, you are right, my comment was indeed incorrect.  Thanks for the comment!
90d250897ec31ecbf9f5a312fd70749ae6c3836b

File: github/resource_github_branch_protection.go (hunk 2 above)
Comment (quantumew) [reply github/resource_github_branch_protection.go]: Hmm looks like that was an old diff, I see it fixed here 96803823fb5504ffe293504b88c8fa3c98b3fd78

**Comment by robsonpeixoto** [comment]
Any news @majormoses ?

**Comment by majormoses** [comment]
> Any news `@majormoses` ?

I am not a maintainer for this project and my golang foo is pretty weak I just reviewed it saying I didnt see anything that looked incredibly stupid.

**Comment by vixus0** [comment]
Any updates on getting this reviewed?

**Comment by quantumew** [comment]
@bflad I noticed you merged the last PR, this has been open for a while, any hope of getting this reviewed? I have been using it and it is quite nice.

**Comment by thatderek** [comment]
Looking forward to seeing this approved.

**Comment by onorua** [comment]
Looking forward to seeing this approved also, this is one of the several issues we need to solve before migrating to terraform GitHub management.

**Comment by acactown** [comment]
Any updates on getting this reviewed?

**Comment by wenbin1989** [comment]
Any updates on getting this reviewed?

**Comment by zhenkaiwang** [comment]
Looking forward to this!

**Comment by paultyng** [comment]
`2.0.0` has been released, including support for this attribute.

**Pull request closed without merging**